### ISSUE TITLE
needless_return: trim suggestion line to avoid pointless indentation

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#4347]
clippy will suggest to remove the "return" here:
````rust
fn main() {
    println!("Hello, world!");
    return;
}
````
when applied by cargo fix however, we will still have the spaces or tabs, which will make for a not-so-pretty diff:
````diff
diff --git a/src/main.rs b/src/main.rs
index de8b85c..991cc4a 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,4 @@
 fn main() {
     println!("Hello, world!");
-    return;
+
 }
````
![indent_line](https://user-images.githubusercontent.com/476013/62580773-582e7680-b8a7-11e9-816c-e16eb5dc8e35.png)

Can we perhaps call .trim() on the suggestion snippet line?

**Comment by killercup** [comment]
I'd be inclined to leave that to rustfmt. Alternatively, find a way to mark suggestions as "should remove full line if only emptiness remains" and also do that for unused imports :)

**Comment by flip1995** [comment]
>  find a way to mark suggestions as "should remove full line if only emptiness remains"

We could add this here:
https://github.com/rust-lang/rust-clippy/blob/7bd7d40789d510eba782fdf86c4a2fdd90084568/clippy_lints/src/utils/diagnostics.rs#L167-L169

But then again, I'm not sure if that's even possible, since `rustfix` just swaps out the span, with the new snippet. So we have to extend the span to the whole line. But only if we made sure, that there's nothing else in that line but whitespace. And also the Suggestion would look really weird then.

Conclusion: I don't think we can fix/influence this in Clippy, since the suggestions are applied by `rustfix`, where we only have influence on with the `span`, but this would make the suggestion look weird..

**Comment by killercup** [comment]
> I don't think we can fix/influence this in Clippy

Yeah, that's what I meant. Add a flag (mark) and handle it in consumers (i.e., rustfix).

**Comment by matthiaskrgr** [comment]
Feel free to close the ticket if you feel this is beyond the scope of clippy, If fully understand.

**Comment by killercup** [comment]
cc @rust-lang/wg-diagnostics -- we should look at this :)
